shbin.py

```diff
@@ -99,21 +115,42 @@ def download(url_or_path, repo, user):
     $ shbin dl https://github.com/Shiphero/pastebin/blob/main/bibo/AWS_API_fullfilment_methods/
     $ shbin dl bibo/AWS_API_fullfilment_methods/
     """
-    path = re.sub(rf"^https://github\.com/{repo.full_name}/(blob|tree)/{repo.default_branch}/", "", url_or_path)
+    full_name_repo = repo.full_name
+    pattern = r"https://github.com/(.*?)/(tree|blob)/"
+    result = re.findall(pattern, url_or_path)
+    if result:
+        full_name_repo = result[0][0]
+
+    is_from_my_repo = "https://" not in url_or_path or full_name_repo == repo.full_name
+    if is_from_my_repo:
+        path = re.sub(rf"^https://github\.com/{repo.full_name}/(blob|tree)/{repo.default_branch}/", "", url_or_path)
+    else:
+        repo = get_repo(full_name_repo)
+        path = get_path(url_or_path)
+
     path = path.rstrip("/")
     try:
-        content = repo.get_contents(path)
-        if isinstance(content, list):
-            # FIXME currently this will flatten the tree:
-            # suposse dir/foo.py and dir/subdir/bar.py
-            # Then `$ shbin dl dir` will get foo.py and bar.py in the same dir.
-            for content_file in content:
-                download(content_file.path, repo, user)
-            return
+        if is_from_my_repo:
+            content = repo.get_contents(path)
+            if isinstance(content, list):
+                # FIXME currently this will flatten the tree:
+                # suposse dir/foo.py and dir/subdir/bar.py
+                # Then `$ shbin dl dir` will get foo.py and bar.py in the same dir.
+                for content_file in content:
+                    download(content_file.path, repo, user)
+                return
+            else:
+                content = content.decoded_content
         else:
-            content = content.decoded_content
+            url = f"https://raw.githubusercontent.com/{repo.full_name}/{path}"
+            response = requests.get(url)
+            if response.status_code > 200:
+                raise Exception("There was a problem with your download, please check url")
+            content = response.content
```

Comment:
what if we unify own or foreign contents with requests? the only difference it that you need to pass the token in the header. 

```python
In [16]: requests.get("https://raw.githubusercontent.com/Shiphero/pastebin/main/mgaitan/15e1VFq87UA.txt",
    ...:     headers={"Authorization": f"token {os.environ['SHBIN_GITHUB_TOKEN']}", "Accept": "application/vnd.github.v3.raw'"}
    ...: )
Out[16]: <Response [200]>

In [17]: Out[16].content[:144]
Out[17]: b"    \n>       mock_sns.push.assert_called_once_with(topic_arn, message)\nE       AssertionError: expected call not found.\nE       Expected: push('"
```

the good news is that passing a token to a public content doesn't affect, so this could be the only code 

```
In [20]: requests.get("https://raw.githubusercontent.com/facebook/react/main/README.md",
    ...:     headers={"Authorization": f"token {os.environ['SHBIN_GITHUB_TOKEN']}", "Accept": "application/vnd.github.v3.raw'"}
    ...: )
Out[20]: <Response [200]>
```